### PR TITLE
Add OIDC user authentication

### PR DESCRIPTION
**Pull request by Wombosvideo**
## New Feature

This pull request adds the following feature(s):

- OIDC user authentication, using [`oic`](https://pypi.org/project/oic/)

## Details of Implementation

- c25c6df: Migrate OIDC integration from <https://github.com/heviat/Mailu-OpenID> to this fork.
- c25c6df and 32422d3: Improve GitHub user experience and add OIDC setup instructions.

### What's new

You can now use any OpenID Connect provider to sign in to your accounts

> [!NOTE]
> This is the initial integration of OIDC in this fork. The documentation for setting it up can be found in the `README.md`

### Breaking changes

_None we know of_

### Previous behavior

Mailu did not support OIDC.

## Checklist

Before we can consider review and merge, please make sure the following list is done and checked.

- [x] Make sure you follow our [Code of Conduct](https://github.com/heviat/Mailu-OIDC/blob/master/CODE_OF_CONDUCT.md).
- [ ] This new feature is tested and works as expected.
- [x] This new feature introduces new functionality[^1].
- [x] This enhancement does not break any existing functionality, or breaks it intentionally (documented above).
- [ ] Unless it's a minor change: Add a [Changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

[^1]: If this pull request enhances existing functionality, please create an enhancement pull request instead. For bug fixes, create a bug-fix pull request instead.

### CODE REVIEW

File: core/admin/mailu/__init__.py [hunk 1]
```diff
@@ -85,6 +89,7 @@ def inject_defaults():
         return dict(
             signup_domains= signup_domains,
             config        = app.config,
+            oic_client    = utils.oic_client,
```
Comment (Wombosvideo) [inline core/admin/mailu/__init__.py:92]: Maybe add a comment here too

File: core/admin/mailu/models.py [hunk 2]
```diff
@@ -20,8 +20,11 @@
 import idna
 import dns.resolver
 import dns.exception
+# [OIDC] Import Flask-Login
+import flask_login
 
 from flask import current_app as app
+from flask import session
```
Comment (Wombosvideo) [inline core/admin/mailu/models.py:27]: Maybe add a comment here too

**Review by Wombosvideo — COMMENTED** [review]
Maybe the two code comments mentioned should be added to better keep track of our changes.

File: core/admin/mailu/__init__.py (hunk 1 above)
Comment (Wombosvideo) [reply core/admin/mailu/__init__.py:93]: Fixed in f73d605

File: core/admin/mailu/models.py (hunk 2 above)
Comment (Wombosvideo) [reply core/admin/mailu/models.py]: Fixed in f73d605

**Review by Encotric — APPROVED** [review]

**Pull request merged**